src/main/java/crawlercommons/sitemaps/sax/XMLHandler.java

```diff
@@ -72,7 +72,20 @@
     public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
         // flush any unclosed or missing URL element
         if (loc.length() > 0 && ("loc".equals(qName) || "url".equals(qName))) {
-            maybeAddSiteMapUrl();
+            // check whether loc isn't white space only
+            for (int i = 0; i < loc.length(); i++) {
```

Comment:
too late. I've merged and I actually meant the opposite as you can see from my subsequent comment.